psychopy/tools/arraytools.py

```diff
@@ -219,6 +219,27 @@ def array2pointer(arr, dtype=None):
         ctypes.POINTER(numpy.ctypeslib.as_ctypes_type(dtype)))
 
 
+def snapto(x, points):
+    """
+    Snap values in array x to their closest equivalent in an array of target values.
+    """
+
+    # Force values to 1d numpy arrays, though keep track of original shape of x
+    x = numpy.array(x)
+    x1d = x.reshape((-1, 1))
+    points = numpy.array(points).reshape((1, -1))
+    # Get differences
+    deltas = numpy.abs(x1d - points)
```

Comment:
This may work without reshaping given numpys broadcasting rules. Just do shape checks before to ensure they are the same.

Reply:
The problem is that I expect them to be different shapes, so the reshaping is necessary. Essentially I need to convert it to 1d to get a 2d matrix of deltas, but once those deltas are converted to linear indices and indexed into `points` we can reshape the resulting array to match the original dimensions of x.

Reply:
Thanks for the documentation, much clearer what I'm looking at.